src/components/PressableWithSecondaryInteraction/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * This is a special Pressable that calls onSecondaryInteraction when LongPressed, or right-clicked.
+ */
+import _ from 'underscore';
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Pressable} from 'react-native';
+
+const propTypes = {
+    onSecondaryInteraction: PropTypes.func.isRequired,
+    children: PropTypes.node.isRequired,
+};
+
+class PressableWithSecondaryInteraction extends Component {
+    constructor(props) {
+        super(props);
+        this.pressableRef = React.createRef();
+    }
+
+    componentDidMount() {
+        this.pressableRef.current.addEventListener('contextmenu', (e) => {
+            e.preventDefault();
+            this.props.onSecondaryInteraction(e);
+        });
+    }
+
+    render() {
+        const defaultPressableProps = _.omit(this.props, ['onSecondaryInteraction', 'children', 'onLongPress']);
```

Comment:
you'll probably catch this but I don't think you want to remove `children` or `onSecondaryInteraction`.

Reply:
Oh I see you're not overwriting props...

Reply:
Hmm I think you can just spread all the props in and then overwrite `onLongPress`

Reply:
But wouldn't you also want to prevent passing the children in? IDK, might not actually do anything but the children shouldn't be passed as props to the `Pressable` right?

Reply:
Is there a difference between

```jsx
<Something>
    {this.props.children}
</Something>
```
and 

```js
<Something children={this.props.children}/>
```

Reply:
downgrading this one to a NAB